applications/browser-extension/src/data/model/Asset.ts

```
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2024 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { type UUID } from "@/types/stringTypes";
+import { validateUUID } from "@/types/helpers";
+import { type components } from "@/types/swagger";
+
+export type Asset = {
+  id: UUID;
+  downloadUrl: URL;
+  filename: string;
+  isUploaded: boolean;
+  updatedAt: Date;
+  createdAt: Date;
+};
+
+export type AssetPreUpload = {
+  asset: Asset;
+  uploadUrl: URL;
+  fields: Record<string, string>;
+};
+
+export function transformAssetResponse(
+  response: components["schemas"]["Asset"],
+): Asset {
+  return {
+    id: validateUUID(response.id),
+    // @ts-expect-error -- TODO: these don't seem to be actually optional, how have we dealt with this in the past?
```

Instead of suppressing errors on all fields, you could use `Required<>` or `SetRequired<>` on the parameter and cast at the call-site with a comment pointing to a TODO

---

Suppressed temporarily here to decide the best course of action, will definitely be handling this without suppressing with a link to a ticket